addons/stock/models/product.py

```diff
@@ -1100,3 +1100,16 @@ def _adjust_uom_quantities(self, qty, quant_uom):
         else:
             computed_qty = self._compute_quantity(qty, procurement_uom, rounding_method='HALF-UP')
         return (computed_qty, procurement_uom)
+
+
+class ProductSupplierInfo(models.Model):
+    _inherit = 'product.supplierinfo'
+
+    delay = fields.Integer(inverse='_inverse_delay')
+
+    def _inverse_delay(self):
+        impacted_orderpoints = self.env['stock.warehouse.orderpoint'].search([
+            ('product_id', 'in', self.product_id.ids),
+        ])
+        impacted_orderpoints.invalidate_recordset(['qty_forecast'])
+        impacted_orderpoints._compute_qty_to_order()
```

Comment:
By looking at the code in 17.2, I fell on this commit: https://github.com/odoo/odoo/commit/19fa747efb272cb4c065069d9652c68bc5694abe I think that adding the `qty_to_order` in the fields to compute as he does here might be more sharper  than calling the compute method dirrectly (if even works in 16.0): 
https://github.com/odoo/odoo/blob/5254b4421537944d201e1cfe434e0d6aa72611a8/addons/stock/models/stock_move.py#L2438-L2441